xds/src/main/java/io/grpc/xds/XdsNameResolver.java

```diff
@@ -246,31 +250,44 @@ public void shutdown() {
         "methodConfig", Collections.singletonList(methodConfig.build()));
   }
 
-  @VisibleForTesting
-  static Map<String, ?> generateServiceConfigWithLoadBalancingConfig(Collection<String> clusters) {
+  // called in syncContext
```

Comment:
I guess if there are multiple callers or the caller is in another Java file, it's fair to use `SynchronizationContext#throwIfNotInThisSynchronizationContext()`, and otherwise it's fine to just have a comment.